### DISCUSSION TITLE
How to access admin console when it's inside of a container?

### DISCUSSION THREAD

**Discussion by dennisjbr** [logto-io/logto#6476] [Q&A]
Hello, I'm having a really hard time getting the initial setup going when logto is in a container.  Ports 3001-3002 are exposed to the host, and the app says it's listening on localhost:3002, and on the admin_login URL.  I set it to be the host-IP-address:3002.   When i tryto hit HTTP://hostip:3002/console or /console/welcome, I just get a blank white page.  Tried on many browsers to no avail.  

Any ideas/help?   thx

**Comment by simeng-li** [comment]
Can you share your container logs?

**Comment by dennisjbr** [comment]
Well it now says :

\npm error command sh -c cd packages/core && NODE_ENV=production node .
\npm notice
npm notice New patch version of npm available! 10.8.1 -> 10.8.2
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.8.2
npm notice To update run: npm install -g ***@***.***
npm notice
\npm error A complete log of this run can be found in: /root/.npm/_logs/2024-08-20T08_35_50_444Z-debug-0.log
\
> start
> cd packages/core && NODE_ENV=production node .

(node:16) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
cache warn No Redis client initialized, skipping
core info Core app is running at http://localhost:3001/
admin info Admin app is running at http://localhost:3002/
---


Because, the only way I could get to the admin console w/o tons of redirects to the original https://login.admin.com domain, I set up an SSH proxy to direct ports 3001 to 3002 to the container.  Then I could get to the site via http://localhost:3002 .  When using “localhost:3002” in the host header and ONLY with that, then the console would show.  Otherwise, no it would just redirect or give a blank page in all of the scenarios I tried.

Thank you!


From: simeng-li ***@***.***>
Date: Tuesday, August 20, 2024 at 2:30 AM
To: logto-io/logto ***@***.***>
Cc: Dennis Brazil ***@***.***>, Author ***@***.***>
Subject: Re: [logto-io/logto] How to access admin console when it's inside of a container? (Discussion #6476)

Can you share your container logs?

**Comment by simeng-li** [comment]
You can customize the endpoints via [environment variables](https://docs.logto.io/docs/references/core/configuration/#environment-variables)

**Comment by dennisjbr** [comment]
I did that too. But it came up with empty white pages. Perhaps I will try one more time with a new container  but the code restrictions of looking only for localhost:3002 is proving to be a challenge specifically when working with docker containers.

Thanks,
-Dennis

On Aug 22, 2024, at 10:36 PM, simeng-li ***@***.***> wrote:

﻿

You can customize the endpoints via environment variables<https://docs.logto.io/docs/references/core/configuration/#environment-variables>
